src/controller/userController.js

```
@@ -176,24 +246,50 @@ class UserController {
   *
  */
   static async unfollowUser(req, res) {
-    const userId = await req.params.id;
-    const followerId = await req.decoded.id;
+    const userId = req.params.id;
+    const followerId = req.decoded.id;
 
-    const isExistingFollowing = await
-    followersUtil.queryForExistingFollowing(true, userId, followerId);
+    try {
+      const userInToken = await findDatabaseField.UserInToken(followerId);
 
-    if (isExistingFollowing) {
-      await followersUtil
-        .queryForUpdatingPreviousFollowing(false, userId, followerId);
-      res.status(200).json({
-        success: true,
-        message: 'You have unfollowed this user'
-      });
-    } else {
-      res.status(400).json({
-        success: false,
-        message: 'You do not follow this user'
-      });
+      if (!userInToken) {
+        return res.status(404).json({ message: 'User does not exist' });
+      }
+
+      const isValidId = await uuidV4Validator(userId);
+
+      if (!isValidId) {
+        return res.status(400).json({
+          success: false,
+          message: 'Invalid Id'
+        });
+      }
+
+      const doesUserInParamsExist = await findDatabaseField
+        .UserInParams(userId, res);
+
+      if (!doesUserInParamsExist) {
+        return res.status(404).json({ message: 'User does not exist' });
+      }
+
+      const isExistingFollowing = await
+      follower.queryForExistingFollowing(true, userId, followerId);
+
+      if (isExistingFollowing) {
+        await follower
+          .queryForUpdatingPreviousFollowing(false, userId, followerId);
+        res.status(200).json({
+          success: true,
+          message: 'You have unfollowed this user'
+        });
+      } else {
+        res.status(400).json({
+          success: false,
+          message: 'You do not follow this user'
+        });
+      }
+    } catch (error) {
+      throw error;
```

Why do you prefer to throw this error, (line 292), and not handle it by sending a response to the user when an error is caught in the catch block?.